dialog/folder.go

```diff
@@ -19,9 +19,27 @@ func NewFolderOpen(callback func(fyne.ListableURI, error), parent fyne.Window) *
 	return dialog
 }
 
-// ShowFolderOpen creates and shows a file dialog allowing the user to choose a folder to open.
+// ShowFolderOpen creates and shows a file dialog allowing the user to choose a
+// folder to open. It takes a callback function that is called whenever the open
+// or cancel buttons are tapped.
```

Comment:
Hmm, I think we might want to word this a bit differently. Technically it takes more than just a callback function.
```suggestion
// folder to open. The callback function will run when the dialog closes.
// The URI will be nil when nothing is selected.
```
I'm not entirely sure about my own suggestion here, but it's just an example (and I couldn't pay attention to wrapping as I wrote it).

Reply:
I'll hold off on any changes until it's decided whether this is the right way to document this, but I feel that to say "close" might still be ambiguous.  

Although "cancel" does close the dialog, I wonder if someone might still miss that "cancel" will still trigger the callback.  Also, from the dialog itself, it doesn't seem like nothing is selected.  The UI makes it looks like if home directory is the default folder.

That's why I tried to be explicit, but it does come with the cost of being more verbose.

Reply:
I think that @Jacalz suggestion here is good, it is quite explicit about the nil which seems to be the main cause for confusion. Of course it should be possible to combine both comments :).